### PR TITLE
Integrate compiletest_rs

### PR DESCRIPTION
**Pull request by fhahn**
This PR integrates `compiletest_rs` as suggested in #5646. I created a new  `tests/compiletest` directory which contains separate crates for the tests.

Currently this PR includes `compile-fail` tests for some lints (acutally all except  inheritance_integrity, beacuse I'm not sure how to include the dom stuff in a way the `#[dom_struct]` works).

I gathered that there should be more crates for which compiletests make sense and would appreciate any pointers to relevant crates.

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/9469)

### CODE REVIEW

**Comment by jdm** [comment]
Let's add `./mach test-compiletest` to `.travis.yml` as well!

**Comment by fhahn** [comment]
Done.

File: python/servo/testing_commands.py [hunk 1]
```diff
@@ -207,6 +210,73 @@ def test_unit(self, test_name=None, package=None):
...
+    @CommandArgument('--release', default=False, action="store_true",
+                     help="Run with a release build of servo")
+    def test_compiletest(self, test_name=None, package=None, release=False):
+        properties = json.loads(subprocess.check_output([
+            sys.executable,
+            path.join(self.context.topdir, "components", "style", "list_properties.py")
+        ]))
+        assert len(properties) >= 100
+        assert "margin-top" in properties
+        assert "margin" in properties
```
Comment (jdm) [inline python/servo/testing_commands.py]: This block isn't relevant to this test harness.

File: python/servo/testing_commands.py [hunk 2]
```diff
@@ -207,6 +210,73 @@ def test_unit(self, test_name=None, package=None):
...
+        self.ensure_bootstrapped()
+
+        if package:
+            packages = {package}
+        else:
+            packages = set()
+
+        test_patterns = []
+        for test in test_name:
+            # add package if 'tests/unit/<package>'
```
Comment (jdm) [inline python/servo/testing_commands.py]: tests/compiletest/

File: python/servo/testing_commands.py [hunk 3]
```diff
@@ -207,6 +210,73 @@ def test_unit(self, test_name=None, package=None):
...
+            match = re.search("tests/compiletest/(\\w+)/?$", test)
+            if match:
+                packages.add(match.group(1))
+            # add package & test if '<package>/<test>', 'tests/unit/<package>/<test>.rs', or similar
```
Comment (jdm) [inline python/servo/testing_commands.py]: tests/compiletest/

**Comment by jdm** [comment]
This is really great work! If you fix the small nits I mentioned, this is ready to merge! Thanks!

**Comment by fhahn** [comment]
Thanks for the feedback! I've pushed a commit which should address it.

**Comment by jdm** [comment]
Looks good! Please squash them and we'll merge your changes.

**Pull request merged**